src/components/Todo.tsx

```diff
@@ -0,0 +1,97 @@
+import { useState } from "react";
+
+interface Todo {
+  id: number;
+  todo: string;
+  completed: boolean;
+}
+
+const Todo = () => {
+  const [id, setId] = useState(0);
+  const [todo, setTodo] = useState<string>("");
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+
+  const handleAdd = () => {
+    
+      const newTodo: Todo = {
+        id: id,
+        todo: todo,
+        completed: false
+      };
+      setTodoList([...todoList, newTodo]);
+      setTodo("");
+      setId(id + 1);
+
+  };
+
+  const deleteTodo = (id: number) => {
+    setTodoList(todoList.filter((todo) => todo.id !== id));
+  };
+
+  const checkCompleted = (id: number) => {
+    setTodoList(
+      todoList.map((todo) =>
+        todo.id === id ? { ...todo, completed: !todo.completed } : todo
+      )
+    );
+  };
+
+  const completedTodos = todoList.filter((todo) => todo.completed);
+  const incompleteTodos = todoList.filter((todo) => !todo.completed);
+
+  return (
+    <div className="todo-container">
+      <h2>Todos</h2>
+      <input
+        type="text"
+        value={todo}
+        placeholder="Enter a todo"
+        onChange={(e) => setTodo(e.target.value)}
```

Comment:
Instead of onChange={(e) => handleInput(e.target.value)}
you can do onChange={handleInput}
& inside handleInput function you can set the value using setTodo

The reason for this is if you use first way i.e your current implementation...It creates a new arrow function reference whenever your component re-renders.
I know it might be confusing for you right now...you will understand it slowly. For now just remember to avoid using arrow functions in JSX